src/digraph/Dominator_Tree.jl

```diff
@@ -0,0 +1,181 @@
+function Dominator_Tree(g::AG, source) where {T, AG<:AbstractGraph{T}}
+
+  parent=zeros(T,nv(g))
+  size=ones(T,nv(g))
+
+  
+  #= the semi dominator is the first elemnt on the path between the root and the node of interst, 
+  such that  there is at least two disjoint paths between it and the node of interst ,
+  if there is not such a node we will consider the parent of the node of interst to be the semi dominator of it
+   there is a duality between the semi dominator and the  immediate dominator 
+   min(preorderof(semi_dominator(v)))>=max(preorder(dominator(v)) 
+   for more details see the paper
+   the key theorm is 
+   Let w != the source and let u be a vertex for which semi_dominator(u) is minimum 
+   among vertices between w and its semi dominator. 
+   Then the immediate dominator is  
+   semi_dominator(w) if semi_dominator(w) = semi_dominator(u), 
+   immidiate_dominator(w) =  immidiate_dominator(u) otherwise.
+  =#
+  semi=zeros(T,nv(g))
+  #==# 
+   
+  #=
+   we will store each node in the bucket of its semi dominator,
+   untill we we figure out all the semi dominator of the nodes between them 
+  =#       
+     
+  buctes=Array{Array,1}(undef,nv(g)) 
```

Comment:
should this be `buckets`?

Also, use `Vector` instead of `Array, 1`.

Right now, this looks like it's an array of vectors. What is its type supposed to be? You've got `eltype(buctes)` as `Array` which is probably not right.